### PR TITLE
DOCSP-43080: Cursors

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-43080>
Staging - <https://deploy-preview-33--docs-c.netlify.app/read/cursors/>

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/read/cursors.txt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: read, results, oplog
+
+Overview
+--------
+
+In this guide, you can learn how to access data from a **cursor** with the
```
Comment (norareidy) [inline source/read/cursors.txt:23]: S: avoid "with"
```suggestion
In this guide, you can learn how to access data from a **cursor** by using the
```

File: source/read/cursors.txt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+.. _c-cursors-iterate:
+
+Access Cursor Contents Iteratively
+----------------------------------
+
+To iterate over the contents of a cursor, use a while loop. The following example retrieves
```
Comment (norareidy) [inline source/read/cursors.txt:47]: S: [nit] I think "while" should be monospaced
```suggestion
To iterate over the contents of a cursor, use a ``while`` loop. The following example retrieves
```

File: source/read/cursors.txt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+.. io-code-block::
+   :copyable:
+
+   .. input:: /includes/read/cursors.c
+      :start-after: start-cursor-iterate
+      :end-before: end-cursor-iterate
+      :language: c
+      :dedent:
+
+   .. output::
```
Comment (norareidy) [inline source/read/cursors.txt:60]: S: these output documents are pretty long - maybe just show the _id, name, and restaurant_id fields, with ... between _id and name?

File: source/read/cursors.txt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+      ...
+
+Retrieve Documents Individually
+-------------------------------
+
+Retrieve documents from a cursor individually by calling the ``mongoc_cursor_next()`` function.
+When calling the ``mongoc_cursor_next()`` function, the function will iterate the cursor and
+set the ``bson`` parameter to the next document in the cursor.
```
Comment (norareidy) [inline source/read/cursors.txt:75]: S: for clarity
```suggestion
This function iterates over the cursor and
sets the ``bson`` parameter to the next document in the cursor.
```

File: source/read/cursors.txt [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+Troubleshooting
+---------------
+
+"*CursorNotFound* cursor id not valid at server"
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cursors in MongoDB can timeout on the server if they've been open for
+a long time without any operations being performed on them. This can
+lead to a ``CursorNotFound`` exception when you try to iterate through the cursor.
```
Comment (norareidy) [inline source/read/cursors.txt:135]: S: could you add another quick sentence about how to troubleshoot this error (I'm guessing just rerun the operation and return a new cursor)?

**Review by norareidy — APPROVED** [review]
LGTM with a few suggestions

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**